### DISCUSSION TITLE
WiFi API Doc Addition

### DISCUSSION THREAD

**Discussion by Xylopyrographer** [espressif/arduino-esp32#7384] [Q&A]
Good day all. With winter in this neck of the woods soon to be upon us, nothing better to do than huddle around the fireplace with a warm beverage and laptop and write documents. 🤪

Considering adding a few things to the [WiFi API](https://docs.espressif.com/projects/arduino-esp32/en/latest/api/wifi.html#ap-example) file. Particularly in fleshing out the examples to demonstrate more of the methods in the libraries and the sequence dependancies.

This'll be for the basic STA and AP stuff to start. So in this vein, I'd like to tap into the brain trust here with a few questions:

1. For AP & STA use, the WiFi API doc should only describe the _differences_ for the ESP32 (and all variants) from the  official [Arduino WiFi file](https://www.arduino.cc/reference/en/libraries/wifi/)?
    - the doc would link back to the official Arduino reference otherwise.
3. Is there a canonical reference for the sequence of calls? For example:
    - in what order should `softAP`, `softAPConfig`, `softAPsetHostname` and `softAPmacAddress` be called?
4. What are the defaults for, well everything, like the AP IP address, gateway, etc.?
5.  If a call is made to `softAPdisconnect`, what then needs to be done to fire up the AP again?
6. Before digging around a swack of header files, is there a reference for what methods are part of which `#include` file?

There will only be a bajillion more questions but that's probably enough to start.

Of course, if someone else has this tucked away in a drawer somewhere or is already on this, glad to collaborate & make it part of the official repo docs.

Thanks.

**Comment by VojtechBartoska** [comment]
Hello @Xylopyrographer,

nice description and approach.

Tagging @pedrominatel, who can help answering some of your questions!
